### PR TITLE
Feature/change cnv con

### PR DESCRIPTION
**Pull request by sickler-alex**
#### What scientific question is your analysis addressing?

[Issue 356](https://github.com/PediatricOpenTargets/ticket-tracker/issues/356) change manta to gatk.

#### What was your approach?

Replace awk command with one that works with the output of gatk cnv and rename manta to gatk.

#### What GitHub issue does your pull request address?

[356](https://github.com/PediatricOpenTargets/ticket-tracker/issues/356)

#### Which areas should receive a particularly close look?

The point of this PR is to replace manta with gatk. The output hasn't been evaluated other than that the snakemake workflow runs successfully.

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by sickler-alex** [comment]
[Issue 357](https://github.com/PediatricOpenTargets/ticket-tracker/issues/357) which deals with the creation of the actual input file for this isn't done yet, so I made this input file with only 5 samples for testing.
[cnv-gatk.seg.gz](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/files/8994173/cnv-gatk.seg.gz)

**Comment by jharenza** [comment]
@sickler-alex I removed review requests until the full file has been run and outputs are in this PR. What do you think your timeline on that will be (ok if not soon-ish since I know you are working on subtyping), but we will be planning two week sprints to start Monday, and just want to know if we should expect this to be reviewed during that time. Also, if you can resolve the conflict as well before the new request for review, that would be great. Thanks!

File: analyses/copy_number_consensus_call/Snakefile [hunk 1]
```diff
@@ -77,33 +77,36 @@ rule cnvkit_filter:
...
-        ## the first awk also filters out for CNV length
+        ## The first awk looks at column 7 to filter out for loss/gain (-/+).
+        ## Then it prints out all 8 of the columns above.
+        ## It writes NA for both p-value and copy number since gatk results don't have these values.
+        ## The first awk command also changes -/+ to DEL/DUP to match the other callers formatting
+        ## The second awk filters the CNV length, and add in the CNV type
         ## The sort command sorts the first digit of chromosome number numerically
-        ## The last pipe is to introduce tab into the file and output file name.        
-        """awk '$6~/DEL/ {{if ($5 > {params.SIZE_CUTOFF} && $11 == 'PASS') {{print "chr"$2,$3,$4,$5,"NA","NA","NA",$6}}}}' {input} """
+        ## The last pipe is to introduce tab into the file and output file name.
```
Comment (migbro) [inline analyses/copy_number_consensus_call/Snakefile:101]: In looking over the .tsv output, I see value `$6`. I don't think that is intended, what is supposed to be there?

**Review by migbro — CHANGES_REQUESTED** [review]
Seems to work for the most part, except for some odd output for GATK in the tsv result file. Is that expected? I recommend doing a spot check of found CNVs in the output to the source file after correcting, if needed, to ensure the translation looks good.

**Review by sakshamphul — COMMENTED** [review]
Looks good!

**Comment by jharenza** [comment]
@kelseykeith will compare the NBL WGS samples using the v11 consensus compared to the consensus output here, using GATK instead of Manta, with a specific look at MYCN amplification. Please also compare these results with the annotation in the histology file for MYCN status (in `molecular_subtype`). The histologies file may not be 100% accurate, but if any changes need to be made, there should only be a handful, and we would go back and do some manual inspection of the CNV plots and/or create those plots.

I will tackle PBTA WGS, with a specific focus on HGG (MET, PDGFRA, TP53, etc), LGG, and ETMRs (c19MC amplification). Will probably re-plot the main oncoprints from OpenPBTA ([here](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/figures/pdfs/fig2/figure2.pdf)).

**Comment by jharenza** [comment]
Hi @sickler-alex. Could you checkout a branch from this branch and run the `focal-cn-preparation` module temporarily changing the `data` dir here 

https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/f8f59e3f3f6379e2fe3b825a67af38473c10b709/analyses/focal-cn-file-preparation/02-add-ploidy-consensus.Rmd#L45-L46) 

to this analysis dir. Then, commit results into a second, stacked PR? We will need the annotated files here for some of the comparisons to be made.

Note: it will have to be run on EC2.

**Comment by sickler-alex** [comment]
This is the new PR with the new results. https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/235

**Comment by kelseykeith** [comment]
Neuroblastoma MYCN looks good! For MYCN for the samples that are in common between the current dev file and the updated file in this PR, as expected the specific consensus segment changes (see first plot below), but the overall MYCN copy number calls are identical (second plot). 

For this plot the darker green shows the overlaps in the segment between the original and the updated consensus segments in this PR, while the light blue is the new PR only while pink would be previous version only (dev).
segments_overlap_not_identical
This plot shows the overlap of ploidy, copy number and status calls between new and old versions. All overlapping neuroblastoma samples have the same call.
mycn_calls_identical
<br>
Also, @jharenza I tried to look at the MYCN status in `pathology_free_text_diagnosis`, but the overlap of samples that were in the old version and had "myc" in their free text was only 3. They don't exactly match, but I'm not sure how amplification is defined in the free text so it may just be a definition problem.
| gene_symbol | biospecimen_id | copy_number.dev | status.dev | copy_number.pr | status.pr |  pathology_free_text_diagnosis |
| -- | -- | -- | -- | -- | -- | -- |
| MYCN | BS_2P6MH05V | 3 | gain | 3 | gain |  MYCN non-amp |
| MYCN | BS_8XDZQKSD | 3 | gain | 3 | gain |  MYCN amp |
| MYCN | BS_V4VGG98Y | 3 | gain | 3 | gain | MYCN non-amp |

**Comment by jharenza** [comment]
Hi @kelseykeith - thanks for this - one quick question, can you change to use green/purple or some other divergent colors?

**Comment by kelseykeith** [comment]
Is this version better? It's hard to visualize because the old segments are primarily within the new segments. If needed I can try and come up with another visualization
segments_overlap_not_identical

**Comment by jharenza** [comment]
>Also, @jharenza I tried to look at the MYCN status in pathology_free_text_diagnosis, but the overlap of samples that were in the old version and had "myc" in their free text was only 3. They don't exactly match, but I'm not sure how amplification is defined in the free text so it may just be a definition problem.

@kelseykeith I didn't catch this entirely. For v10, MYCN status would be in the `molecular_subtype` column. For v11, it was transferred to the `pathology_free_text_diagnosis` column. Can you do the comparison of the subtypes using this data? Then, I think we can call it a wrap. Thanks!

**Comment by kelseykeith** [comment]
I'll take a look at it again, sure!

**Comment by kelseykeith** [comment]
So the both the old (dev) and new (pr207) CNV consensus calls have the same copy number and status for MYCN in the Neuroblastoma samples that are in both v10 and v11. However, in 10 / 70 samples the copy number call doesn't quite agree with the MYCN amplification status from v10, see the samples in the table below, where the columns for the previous version have the ending .dev and columns for the new version created in this PR have the ending .pr207. I'm not sure how MYCN status was detemined for the `molecular_subtype` in the v10 histology, but most of the cases are borderline, so it could be a difference in definitions. The only one that seems like an issue is the last one in the table where we called a copy number of 30 but the MYCN status was `non-amp`.

| biospecimen_id | status.dev | copy_number.dev | molecular_subtype.dev | status.pr207 | copy_number.pr207 | pathology_free_text_diagnosis.pr207 |
| --- | --- | --- | --- | --- | --- | --- |
|BS_4J8JK335 | amplification | 5 | MYCN non-amp| amplification | 5 | NA |	
| BS_6K2SSJ77 | gain | 3 | MYCN amp | gain | 3 | NA |
| BS_4WSW3X49 | gain | 3 | MYCN amp | gain | 3 | NA |
| BS_ACCE0MEA | gain | 3 | MYCN amp | gain | 3 | NA |
| BS_8XDZQKSD | gain | 3 | MYCN amp | gain | 3 | NA |
| BS_E4V8XN3G | gain | 3 | MYCN amp | gain | 3 | NA |
| BS_FJP95JXF | gain | 3 | MYCN amp | gain | 3 | NA |
| BS_QDRHQSPZ | gain | 3 | MYCN amp | gain | 3 | NA |
| BS_WMM0MP1J | gain | 4 | MYCN amp | gain | 4 | NA |
| BS_ZED6T1CT | amplification| 30 | MYCN non-amp | amplification | 30 | NA |

Also, mostly unrelated, in the v11 histology file `pathology_free_text_diagnosis` does not have information about the MYCN amplification status for the majority of Neuroblastoma samples, which was what I was trying to say in my previous comment.

**Comment by jharenza** [comment]
>So the both the old (dev) and new (pr207) CNV consensus calls have the same copy number and status for MYCN in the Neuroblastoma samples that are in both v10 and v11. 

OK, based on this, I think that we have enough information and we will update v12 to add the missing MYCN annotations. cc @ewafula

**Review by jharenza — DISMISSED** [review]
@sickler-alex after GA passes, feel free to merge this next

File: analyses/focal-cn-file-preparation/02-add-ploidy-consensus.Rmd [hunk 2]
```diff
@@ -42,7 +42,7 @@ if(!dir.exists(output_dir)) {
 
 ```{r}
 # read in the histology and consensus seg file from the data directory
-consensus_seg_file <- file.path("..", "..", "data", 
+consensus_seg_file <- file.path("..", "copy_number_consensus_call", "results",
```
Comment (jharenza) [inline analyses/focal-cn-file-preparation/02-add-ploidy-consensus.Rmd:45]: Note that we will have to switch this back once v12 is released

File: analyses/copy_number_consensus_call/run_consensus_call.sh [hunk 3]
```diff
@@ -16,11 +16,11 @@ SCRATCHDIR=../../scratch/copy_consensus
 mkdir -p $SCRATCHDIR
 mkdir -p results
 
-## Run the python script to go from 1 big manta file, cnvkit file and freec file into 3 directories. 
+## Run the python script to go from 1 big gatk file, cnvkit file and freec file into 3 directories.
 ## Each directory with individual sample files.
 
 python3 scripts/merged_to_individual_files.py \
-    --manta ../../data/sv-manta.tsv.gz \
+    --gatk ../../data/merged_gatk_cnv_called_copy_ratio_segments_tumor.seg \
```
Comment (jharenza) [inline analyses/copy_number_consensus_call/run_consensus_call.sh]:
```suggestion
    --gatk ../../data/cnv-gatk.seg.gz \
```
let's use this as the filename to be consistent with the other cnv files. @migbro @dmiller15 and/or @zhangb1 - can you confirm the format used will be our standard merged format (I think this file is only local to you @sickler-alex)? @sickler-alex can you point us to this file on cavatica?

**Review by jharenza — COMMENTED** [review]
@sickler-alex before we merge, I made a few minor comments - Can you take a look at the updates I made to the README and make sure I did not miss anything?

File: analyses/copy_number_consensus_call/run_consensus_call.sh (hunk 3 above)
Comment (sickler-alex) [reply analyses/copy_number_consensus_call/run_consensus_call.sh]: https://cavatica.sbgenomics.com/u/kfdrc-harmonization/sd-8y99qzjj-ad-hoc-gatk-cnv-wxs/files/62d045d008804a1f3133d7d2/

Comment (zhangb1) [reply analyses/copy_number_consensus_call/run_consensus_call.sh]: `cnv-gatk.seg.gz ` is good I think ,with other we use `cnv-cnvkit.seg.gz
cnv-controlfreec.tsv.gz`. for cnvkit and controlfreec.....  but for the format.. I am not sure ,, I don't have the answer ..

**Review by jharenza — APPROVED** [review]

**Comment by jharenza** [comment]
Just so we have everything recorded - @dmiller15 or @sickler-alex can one of you describe the GATK CNV methods in a comment below, including version, how it was run (eg - PONs), any non-default params, so we have for a future manuscript while we figure out where to put said manuscript. Thank you!

**Comment by migbro** [comment]
Actually running the workflow and creating the PON, I think that info @zhangb1 has or @bmennis

**Comment by dmiller15** [comment]
The GATK CNV workflow is using GATK version 4.2.4.1. 

As for the methods, a generalized description of the steps is provided in the documentation:
- [PON Creation](https://github.com/kids-first/kf-somatic-workflow/blob/master/docs/kfdrc_gatk_create_cnv_pon_wf.md)
- [Somatic Pair WF](https://github.com/kids-first/kf-somatic-workflow/blob/master/docs/kfdrc_gatk_cnv_somatic_pair_wf.md)

The steps of these workflows are fully described in the GATK documentation: https://gatk.broadinstitute.org/hc/en-us/articles/360035531152--How-to-Call-common-and-rare-germline-copy-number-variants

**Comment by bmennis** [comment]
@jharenza What @dmiller15 provided for the suggested inputs is accurate to what was run.

**Pull request merged**